hs_rest_api2/metadata.py

```diff
@@ -0,0 +1,238 @@
+
+
+@swagger_auto_schema(method='put', request_body=serializers.ResourceMetadataInSerializer,
+                     operation_description="Update Resource level metadata in json",)
+@swagger_auto_schema(method='get', responses={200: serializers.ResourceMetadataSerializer},
+                     operation_description="Get Resource level metadata json")
+@api_view(['GET', 'PUT'])
+def resource_metadata_json(request, pk):
+    if request.method == 'GET':
+        resource, _, _ = authorize(request, pk,
+                                   needed_permission=ACTION_TO_AUTHORIZE.VIEW_RESOURCE)
+        md_json = _resource_metadata_json(resource)
+        return JsonResponse(md_json)
+
+    resource, _, _ = authorize(request, pk,
+                               needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+    md = request.data
+    ingest_resource_metadata(resource, md)
+    return HttpResponse(status=200)
+
+
+def get_aggregation(resource, file_path):
+    file_storage_path = os.path.join(resource.file_path, file_path)
+    folder, file_name = ResourceFile.resource_path_is_acceptable(resource, file_storage_path, test_exists=True)
+    res_file = ResourceFile.get(resource, file_name, folder)
+    return res_file.logical_file
+
+
+def aggregation_metadata(resource, file_path):
+    agg = get_aggregation(resource, file_path)
+    if agg.metadata.is_dirty:
+        # TODO, shouldn't have to do this?
+        agg.create_aggregation_xml_documents()
+    return load_metadata(resource.get_irods_storage(), agg.metadata_file_path)
+
+
+def aggregation_metadata_json_loads(resource, file_path):
+    return json.loads(aggregation_metadata(resource, file_path).json())
+
+
+def get_in_schema(out_schema):
+    # TODO update hsmodels in schemas to include forbid
+    in_schema = out_schema.__bases__[0]
+
+    class IncomingForbid(in_schema):
+        class Config:
+            extra = "forbid"
+    return IncomingForbid
+
+
+def ingest_aggregation_metadata(resource, incoming_metadata, file_path):
+    from hsmodels.schemas import rdf_graph
+
+    # read existing metadata from file
+    agg_md = aggregation_metadata(resource, file_path)
+    agg_md_dict = agg_md.dict()
+
+    # get schema classes
+    out_schema = agg_md.__class__
+    in_schema = get_in_schema(out_schema)
+
+    # validate incoming metadata against the schema
+    try:
+        incoming_md = in_schema(**incoming_metadata)
+    except PydanticValidationError as e:
+        raise ValidationError(e)
+
+    # merge existing metadata with incoming, incoming overrides existing
+    incoming_dict = {**incoming_md.dict(exclude_defaults=True)}
+    existing_dict = {**agg_md_dict}
+    merged_metadata = {**existing_dict,
+                       **incoming_dict}
+
+    agg_metadata = out_schema(**merged_metadata)
+    graph = rdf_graph(agg_metadata)
+
+    # write the updated metadata back to file
+    ingest_logical_file_metadata(graph, resource)
+    create_bag_metadata_files(resource)
```

Comment:
This shouldn't be happening here, we really just need to create the XML files for the updated metadata.

Reply:
fixed